tasks/system_package_management.yml

```diff
@@ -0,0 +1,70 @@
+---
+# .. vim: foldmarker=[[[,]]]:foldmethod=marker
+
+- name: Get ownCloud APT OpenPGP key
+  apt_key:
+    url: '{{ owncloud__apt_repo_key_url }}'
```

Comment:
I think that GPG keys should be pulled from a different source than the APT repository information, that way the trust can be estabilished independently from the source.

I would try and convince the ownCloud authors to publish their GPG keys on the keyservers. They are already in circulation, publication in the OpenPGP keyserver network will make access to them easier not only for DebOps, but for other projects that wish to authenticate the packages as well.

If they won't do it, then yes, GPG keys should be included in the role to provide independent verifable source of the key material. At least this sounds better than downloading the GPG key from a random server on the Internet which can vanish at any moment.

The key should be copied from the role's `files/` directory to the remote host to say, `/usr/local/src/owncloud/` and installed via `apt_key` module. This should ensure idempotent operation.